libpromises/policyserver.c

```diff
@@ -0,0 +1,300 @@
+                                            "policy_hub_ip" );
+        EvalContextVariableRemoveSpecial(   ctx, SPECIAL_SCOPE_SYS,
+                                            "policy_hub_port" );
+        return;
+    }
+
+    // Set POLICY_SERVER full string:
+    POLICY_SERVER = xstrdup(new_policy_server);
+
+    // Parse new_policy_server and set POLICY_SERVER_IP + POLICY_SERVER_PORT
+    {
+        // Parse policy server in a separate buffer:
+        char *host, *port;
+        char buffer[CF_MAX_SERVER_LEN];
+        strncpy(buffer, new_policy_server, CF_MAX_SERVER_LEN);
+        ParseHostPort(buffer, &host, &port);
+
+        // Set POLICY_SERVER_HOST and POLICY_SERVER_PORT:
+        POLICY_SERVER_HOST = xstrdup(host);
+        if (port == NULL)
+        {
+            strncpy(POLICY_SERVER_PORT, CFENGINE_PORT_STR, CF_MAX_PORT_LEN);
+        }
+        else
+        {
+            strncpy(POLICY_SERVER_PORT, port, CF_MAX_PORT_LEN);
+        }
+
+    }
+
+    EvalContextVariablePutSpecial(  ctx,  SPECIAL_SCOPE_SYS,
+                                    "policy_hub", POLICY_SERVER_HOST,
+                                    CF_DATA_TYPE_STRING,
+                                    "source=bootstrap" );
+
+    // Doo lookup and set POLICY_SERVER_IP:
+    int ret = Hostname2IPString(POLICY_SERVER_IP, POLICY_SERVER_HOST,
+                                CF_MAX_IP_LEN);
+
+    // Set policy variable sys.policy_hub_ip:
+    if (ret == 0)
+    {
+        EvalContextVariablePutSpecial(  ctx,  SPECIAL_SCOPE_SYS,
+                                        "policy_hub_ip", POLICY_SERVER_IP,
+                                        CF_DATA_TYPE_STRING,
+                                        "derived-from=sys.policy_hub" );
+    }
+    else // IP Lookup failed
+    {
+        EvalContextVariableRemoveSpecial(   ctx, SPECIAL_SCOPE_SYS,
+                                            "policy_hub_ip" );
+        POLICY_SERVER_IP[0] = '\0';
+    }
+
+    // Set the sys.policy_hub_port variable:
+    if (POLICY_SERVER_PORT[0] != '\0')
+    {
+        EvalContextVariablePutSpecial( ctx, SPECIAL_SCOPE_SYS,
+                                       "policy_hub_port", POLICY_SERVER_PORT,
+                                       CF_DATA_TYPE_STRING,
+                                       "source=bootstrap" );
+    }
+    else // Default value (5308) is set
+    {
+        EvalContextVariablePutSpecial( ctx, SPECIAL_SCOPE_SYS,
+                                       "policy_hub_port",
+                                       CFENGINE_PORT_STR,
+                                       CF_DATA_TYPE_STRING,
+                                       "source=bootstrap" );
+    }
+}
+
+
+//*******************************************************************
+// POLICY SERVER GET FUNCTIONS:
+//*******************************************************************
+
+// TODO: Shouldn't be necessary? Can be inline too
+static const char* PolicyServerCheckedReturn(char* rval)
```

Comment:
Don't use the word `rval` for a string, `Rval` is a very widely type used in the CFEngine source code so it creates confusion. Use `host` or `ip` or whatever this is, or even `s` is better in this context since it's simple.